service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sagemakerNotebook/DeleteAwsSagemakerNotebookStep.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.workspace.service.resource.controlled.cloud.aws.sagemakerNotebook;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.exception.InternalLogicException;
+import bio.terra.workspace.common.utils.AwsUtils;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+
+public class DeleteAwsSagemakerNotebookStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteAwsSagemakerNotebookStep.class);
+
+  private final ControlledAwsSagemakerNotebookResource resource;
+  private final AwsCloudContextService awsCloudContextService;
+
+  public DeleteAwsSagemakerNotebookStep(
+      ControlledAwsSagemakerNotebookResource resource,
+      AwsCloudContextService awsCloudContextService) {
+    this.resource = resource;
+    this.awsCloudContextService = awsCloudContextService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    AwsCredentialsProvider credentialsProvider =
+        AwsUtils.createWsmCredentialProvider(
+            awsCloudContextService.getRequiredAuthentication(),
+            awsCloudContextService.discoverEnvironment());
+
```

Comment:
Is my assumption right that if the notebook is still running, this will fail and the CLI/UI will surface this as a "user actionable" error?